src/Sylius/Bundle/CoreBundle/DependencyInjection/SyliusCoreExtension.php

```diff
@@ -70,24 +73,43 @@ public function prepend(ContainerBuilder $container): void
         $config = $container->getExtensionConfig($this->getAlias());
         $config = $this->processConfiguration($this->getConfiguration([], $container), $config);
 
+        $this->prependSyliusThemeBundle($container, $config['driver']);
+        $this->prependHwiOauth($container);
+        $this->prependDoctrineMigrations($container);
+    }
+
+    private function prependHwiOauth(ContainerBuilder $container): void
+    {
+        $loader = new XmlFileLoader($container, new FileLocator(__DIR__ . '/../Resources/config'));
+
+        if (!$container->hasExtension('hwi_oauth')) {
+            return;
+        }
+
+        $loader->load('services/integrations/hwi_oauth.xml');
+    }
+
+    private function prependSyliusThemeBundle(ContainerBuilder $container, string $driver): void
+    {
         foreach ($container->getExtensions() as $name => $extension) {
             if (in_array($name, self::$bundles, true)) {
-                $container->prependExtensionConfig($name, ['driver' => $config['driver']]);
+                $container->prependExtensionConfig($name, ['driver' => $driver]);
             }
         }
 
         $container->prependExtensionConfig('sylius_theme', ['context' => 'sylius.theme.context.channel_based']);
-
-        $loader = new XmlFileLoader($container, new FileLocator(__DIR__ . '/../Resources/config'));
-        $this->prependHwiOauth($container, $loader);
     }
 
-    private function prependHwiOauth(ContainerBuilder $container, LoaderInterface $loader): void
+    private function prependDoctrineMigrations(ContainerBuilder $container): void
     {
-        if (!$container->hasExtension('hwi_oauth')) {
+        if (!$container->hasExtension('doctrine_migrations')) {
             return;
         }
 
-        $loader->load('services/integrations/hwi_oauth.xml');
+        $container->prependExtensionConfig('doctrine_migrations', [
+            'migrations_paths' => [
+                'Sylius\Bundle\CoreBundle\Migrations' => __DIR__ . '/../Migrations',
```

Comment:
as of https://github.com/doctrine/DoctrineMigrationsBundle/pull/318 you can use also bundle relative paths

Reply:
Is it the feature above the main reason why you have added these mapping dynamically? Cleanup of this class could be merged anyway

Reply:
Yeah, it could be moved to regular config files we provide with CoreBundle now.